react/BindingChallenge.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react'
+import { ButtonWithIcon, IconClose } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useDevice } from 'vtex.device-detector'
+import { useCssHandles } from 'vtex.css-handles'
+
+import { useBinding } from './hooks/useBindings'
+import BindingSelectorBlock from './BindingSelectorBlock'
+
+const close = <IconClose />
+
+const CSS_HANDLES = ['challengeBar', 'actionCTA', 'actionContainer'] as const
+
+interface Props {
+  barText: string
+}
+
+const BindingChallenge = ({ barText }: Props) => {
+  const [show, setDisplay] = useState(false)
+  const [viewerCountry, setViewerCountry] = useState<string | null>(null)
+  const handles = useCssHandles(CSS_HANDLES)
+  const { device } = useDevice()
+
+  const {
+    data: { currentBinding, bindingsError, loadingBindings },
+  } = useBinding()
+
+  useEffect(() => {
+    fetch('/_v/binding-selector/viewer-country').then((response) =>
+      response.text().then((countryISO: string | null) => {
+        !countryISO || setViewerCountry(countryISO)
+      })
+    )
+  }, [])
+
+  useEffect(() => {
+    if (!viewerCountry || !currentBinding.defaultLocale) {
+      return
+    }
+
+    if (localStorage.getItem('hasShownBindingBar') === 'true') {
+      return
+    }
+
+    /* viewer country is ISO2
+    so we compare the last 2 digits of the locale
+    i.e es-ES === ES
+    */
+    const bindingLocale = currentBinding.defaultLocale
+      ?.substr(3, 2)
+      .toUpperCase()
+
+    if (viewerCountry === bindingLocale) {
+      return
+    }
+
+    setDisplay(true)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think we can remove this eslint flag. I guess all variables are listed in the deps.

Reply:
Removed